virt-v2v/cold/entrypoint.go

```diff
@@ -114,7 +116,14 @@ func buildCommand() []string {
 		fmt.Println("Error creating directory  ", err)
 		os.Exit(1)
 	}
-	virtV2vArgs = append(virtV2vArgs, "-o", "local", "-os", DIR)
+	virtV2vArgs = append(virtV2vArgs, "-o", "kubevirt")
+
+	if checkEnvVariablesSet("V2V_NewName") {
+		virtV2vArgs = append(virtV2vArgs, "-on", os.Getenv("V2V_NewName"))
+		nameChanged = true
+	}
```

Comment:
what's the benefit letting virt-v2v change the VM name during conversion?

Reply:
It doesn't especially make any different for `-o kubevirt` mode, but if you're interested in what it actually affects, search for `output_name` in these two files:
https://github.com/libguestfs/virt-v2v/blob/master/output/create_kubevirt_yaml.ml
https://github.com/libguestfs/virt-v2v/blob/master/output/output_kubevirt.ml

Reply:
Thanks @rwmjones.
Based on what I see, the main reason for it is when we have a name of the RFC1123 standard which cause an error in the YAML creation(the same reason we change it on our end).
@bkhizgiy can you please add a comment on this and within the commit?

Reply:
Yes, as @liranr23 mentioned, since the output is in kubevirt format, it should meet the k8s DNS1123 RFC requirements. The creation of the produced VM YAML fails when using an unsupported name. We are already changing the name in that case, and additionally propagating the value to virt-v2v to overcome this.

And sure, I'll add a comment.

Reply:
Is that a bug in virt-v2v, eg is it generating the yaml quoting wrongly?  Here's that code: https://github.com/libguestfs/virt-v2v/blob/e9c0d63b22bcb4c31cfaac00fc1271713c8bd3b3/lib/YAML.ml#L126

Reply:
@rwmjones Nope, it's the desired behavior. :) There is a validation on the name before the YAML creation. Otherwise, what will happen is that once we try to apply the YAML with the original name, the VM creation will fail because the name is illegal.